Test/Test-c/test.c

```diff
@@ -0,0 +1,92 @@
+//
```

Comment:
The CI_Build workflow already contains a test that solves an instance and compares solution values. 
In addition, within the Julia and Python interface repositories, you have included tests that cover all the intermediate steps: Julia/Python => Calling these C functions from the lib => Solving. Perhaps adding this Test folder here contributes to a bit more clutter (requires to discuss more options in the README + additional CMake options things etc..) without many benefits. Finally, including real unit tests in this folder would take a lot of development, I don't believe we will be needing them. What is your opinion?

Reply:
Your observation about Julia/Python is correct: they also test the solutions. 

The reason I included C function tests here is as follows.  While developing Python/Julia packages, if the connection to the HGS-CVRP solver is not working correctly, it is not sure if the problem occurs in the C function parts of the HGS, or it occurs in how the shared library of HGS is installed and connected within Python/Julia.  So, by testing C functions here within HGS-CVRP, we can eliminate one source of issues. 

FYI, the Python/Julia packages have multiple layers. When one installs the package, the following steps will happen:
1. To download HGS-CVRP source files from this Github repository. 
2. To compile a shared library from the source with C functions.
3. To install the compiled shared library in an appropriate place.

After that, one imports the package to Python/Julia, it loads the installed shared library into the library and connects to it. The connection itself isn't that straightforward, since we need to match the argument types and return types, without using the header files. So, any mistake can happen.

So, for debugging purposes, we need to make sure that the C interface works fine, before we jump into Python/Julia packages. This is also why I have a habit of including `hello()` functions :)   This is a good test function to confirm the connection between Python/Julia and HGS is normal.  But with the C function tests, I think we don't need the hello function.

Reply:
Sure, I now better understand the reasoning behind these additional tests. Let's keep them inside then. If you agree, I will proceed with the merge of the pull request.

Reply:
In the README, we can let the C++ users of HGS-CVRP test only the necessary part. 

For example, to build and test the executable:
```
make bin
ctest -R bin
```
This will build `./hgs` and test the two instances using the binary file.


To build and test the library:
```
make lib
ctest -R lib
```
This will create `libhgscvrp.so` and test the C functions.

Reply:
@vidalt Great! Yes, please go ahead and merge.  I will simplify the make and ctest processes and work on the README file later.

Reply:
Sounds good!